### PR TITLE
refactor: matching dfn and module filenames

### PR DESCRIPTION
**Pull request by wpbonelli**
* reopen #1597 
* first step toward #1491
* drop numbers from filenames and update naming as necessary so DFNs and corresponding module files match &mdash; paired component files are named `<dfn name>idm.f90`
* makes explicitly mapping DFN to paired IDM file unnecessary: #1603
* applies to files in
  * `src/Exchange`
  * `src/Model/GroundWaterFlow`
  * `src/Model/GroundWaterTransport`
  * `src/Model/GroundWaterEnergy`
  * `src/Model/SurfaceWaterFlow`
  * `src/Model/TransportModel`
  * `src/Solution/LinearMethods`

### CODE REVIEW

File: autotest/test_swf_dfw_gwf.py [hunk 1]
```diff
@@ -58,20 +58,20 @@ def build_models(idx, test):
     add_swf_model(sim)
     add_gwf_model(sim)
 
-    swfgwf_data = [
+    gwfswf_data = [
```
Comment (langevin-usgs) [inline autotest/test_swf_dfw_gwf.py:61]: Is there are a reason for changing this?  My thinking is that the name should be swf-gwf to loosely support the idea of upper to lower.  This order also corresponds to the cellids in the swfgwf_data list.  The first tuple there is a SWF reach number and the second is the GWF cellid.

File: src/Model/GroundWaterFlow/gwf.f90 [hunk 2]
```diff
@@ -1175,7 +1175,7 @@ end subroutine gwf_da
   !> @brief GroundWater Flow Model Budget Entry
   !!
   !! This subroutine adds a budget entry to the flow budget.  It was added as
-  !! a method for the gwf3 model object so that the exchange object could add its
+  !! a method for the gwf- model object so that the exchange object could add its
```
Comment (langevin-usgs) [inline src/Model/GroundWaterFlow/gwf.f90:1178]:
```suggestion
  !! a method for the gwf model object so that the exchange object could add its
```

**Review by langevin-usgs — COMMENTED** [review]
Hey @wpbonelli, this is great.  I think we should keep the swfgwf name, but let me know if there is reason to change it.

File: autotest/test_swf_dfw_gwf.py (hunk 1 above)
Comment (wpbonelli) [reply autotest/test_swf_dfw_gwf.py:61]: Understood. I had two reasons for doing this: consistency with GWT, GWE (and PRT), and it makes [the logic in `src/meson.build`](https://github.com/MODFLOW-USGS/modflow6/pull/1532/files#diff-f28598af2e23aa5d2bc7c72e022ae2c56a33802eb970afffaeca1e40607f97feR27) to programmatically define exchanges slightly simpler if we can assume the order &mdash; but can just check the `src/Exchange` directory instead to see which exchanges are present. I will restore the original swf-gwf namings.

**Review by langevin-usgs — APPROVED** [review]

**Pull request merged**